mu/logic.py

```diff
@@ -1074,28 +1096,51 @@ def show_admin(self, event=None):
 
         Ensure any changes to the envars is updated.
         """
-        logger.info('Showing logs from {}'.format(LOG_FILE))
+        logger.info('Showing admin with logs from {}'.format(LOG_FILE))
         envars = '\n'.join(['{}={}'.format(name, value) for name, value in
                             self.envars])
         settings = {
             'envars': envars,
             'minify': self.minify,
             'microbit_runtime': self.microbit_runtime,
         }
+        packages = installed_packages()
         with open(LOG_FILE, 'r', encoding='utf8') as logfile:
-            new_settings = self._view.show_admin(logfile.read(), settings)
-            self.envars = extract_envars(new_settings['envars'])
-            self.minify = new_settings['minify']
-            runtime = new_settings['microbit_runtime'].strip()
-            if runtime and not os.path.isfile(runtime):
-                self.microbit_runtime = ''
-                message = _('Could not find MicroPython runtime.')
-                information = _("The micro:bit runtime you specified ('{}') "
-                                "does not exist. "
-                                "Please try again.").format(runtime)
-                self._view.show_message(message, information)
+            new_settings = self._view.show_admin(logfile.read(), settings,
+                                                 '\n'.join(packages))
+            if new_settings:
+                self.envars = extract_envars(new_settings['envars'])
+                self.minify = new_settings['minify']
+                runtime = new_settings['microbit_runtime'].strip()
+                if runtime and not os.path.isfile(runtime):
+                    self.microbit_runtime = ''
+                    message = _('Could not find MicroPython runtime.')
+                    information = _("The micro:bit runtime you specified "
+                                    "('{}') does not exist. "
+                                    "Please try again.").format(runtime)
+                    self._view.show_message(message, information)
+                else:
+                    self.microbit_runtime = runtime
+                new_packages = [p for p in
+                                new_settings['packages'].lower().split('\n')
+                                if p.strip()]
```

Comment:
If only the `new_packages` list is lower cased (and `packages` left with mixed case) you can end up with false positives when the two lists are compared in `self.sync_package_state()`.

Reply:
The convention from Python packages means the result of `installed_packages` stored in `packages` will always contain lower-case package names.

Reply:
Pygments appears in that list with a capital letter, so if I guess the standard isn't enforced, or something else is going on.

Reply:
So I've just tried installing and removing Pygmants. It works fine. 

I'm not sure what notion of a "false positive" you're getting at here. Looking at the code, the set operations and the way the directories ending in `dist-info` are used, I honestly don't believe this is a problem. 

Please can you give me a **concrete** example of where a failure occurs.

Reply:
By concrete I mean repeatable steps I can follow to observe the false positive. Otherwise I'm investing lots of time in trying (and guessing) to recreate the "problem state", and that's no fun for anyone. :-(

Reply:
I've created this GIF (big 4Mbs, better viewed in its own tab), hope it makes things more clear, please do let me know if it doesn't and I'll provide more detailed steps.

![2019-02-12 15 35 19](https://user-images.githubusercontent.com/4189262/52647289-58435000-2edc-11e9-82cb-b590d813c847.gif)

In essence 'Pygments' is constantly uninstalled and reinstalled every time the settings window is closed because of the capitalisation. So the package can be installed and removed correctly, but every time the list is checked it will go through that cycle.

Reply:
@carlosperate you :star: ... Now I **SEE** what you mean. Thank you. I'll fix this tomorrow (dad-duty beckons)..!

Reply:
OK... fixed with latest push. Please re-test. Thank you! (Now to try to fix the damn builds)